homeassistant/components/schlage/lock.py

```diff
@@ -48,6 +48,32 @@ def _update_attrs(self) -> None:
         """Update our internal state attributes."""
         self._attr_is_locked = self._lock.is_locked
         self._attr_is_jammed = self._lock.is_jammed
+        # Only update changed_by if we get a valid value. This way a previous
+        # value will stay intact if the latest log message isn't related to a
+        # lock state change.
+        if changed_by := self._get_changed_by():
+            self._attr_changed_by = changed_by
+
+    def _get_changed_by(self):
+        if not (logs := self._lock_data.logs):
+            return None
+
+        want_prefix = "Locked by " if self._lock.is_locked else "Unlocked by "
+        want_prefix_len = len(want_prefix)
+        for log in sorted(logs, reverse=True, key=lambda log: log.created_at):
```

Comment:
It would be nicer if this was all abstracted away in the library so HA doesn't need to know anything about the details of the api implementation

Reply:
Done.